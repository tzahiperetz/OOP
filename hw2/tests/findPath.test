# findpath test

# Create a graph
CreateGraph g1

# Add nodes
CreateNode a 2
CreateNode b 1
CreateNode c 3
CreateNode d 1
CreateNode e 4

AddNode g1 a
AddNode g1 b
AddNode g1 c
AddNode g1 d
AddNode g1 e

# Add an edge
AddEdge g1 a b
AddEdge g1 a c
AddEdge g1 b c
AddEdge g1 b d
AddEdge g1 c d
AddEdge g1 c e
AddEdge g1 d e

FindPath g1 a -> e
FindPath g1 b -> c
FindPath g1 a b -> e
FindPath g1 a  -> b c
FindPath g1 a  -> c d
FindPath g1 a b -> e d
