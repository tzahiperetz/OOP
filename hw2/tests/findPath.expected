# findpath test

# Create a graph
created graph g1

# Add nodes
created node a with cost 2
created node b with cost 1
created node c with cost 3
created node d with cost 1
created node e with cost 4

added node a to g1
added node b to g1
added node c to g1
added node d to g1
added node e to g1

# Add an edge
added edge from a to b in g1
added edge from a to c in g1
added edge from b to c in g1
added edge from b to d in g1
added edge from c to d in g1
added edge from c to e in g1
added edge from d to e in g1

shortest path in g1: a b d e
shortest path in g1: b c
shortest path in g1: b d e
shortest path in g1: a b
shortest path in g1: a b d
shortest path in g1: b d
